### PR TITLE
Use HTTPS for docs.astropy.org URLs

### LINKED ISSUE #610: Search on main page results in warning because of use of http
I'm using firefox on a mac m1, when I visit:
https://www.astropy.org/
and use the search box I am presented with a warning:
Screenshot 2024-07-16 at 4 45 20 PM
due to the use of `http` instead of `https` in the search.
https://github.com/astropy/astropy.github.com/blob/745c64e0a6dabd11a5b73de1ad0c083a9585ab33/index.html#L94

### PR DESCRIPTION
**Pull request by pllim**
Fix #610

Just a blind search and replace: `find ./ -type f -exec sed -i 's/http:\/\/docs\.astropy/https:\/\/docs\.astropy/g' {} \;`

### CODE REVIEW

File: announcements/release-0.4.html [hunk 1]
```diff
@@ -76,13 +76,13 @@ <h1>Astropy v0.4 Released!</h1>
 In addition, hundreds of smaller improvements and fixes have been made. An overview of the changes is provided at:
 </p>
 <p>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="http://docs.astropy.org/en/stable/whatsnew/0.4.html">http://docs.astropy.org/en/stable/whatsnew/0.4.html</a>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="https://docs.astropy.org/en/stable/whatsnew/0.4.html">https://docs.astropy.org/en/stable/whatsnew/0.4.html</a>
```
Comment (hamogu) [inline announcements/release-0.4.html:79]: This one gives me a 404 when I try either the old or the new version. So, not a problem for this PR (was broken before I broken now).

**Review by hamogu — APPROVED** [review]
I checked a few links manually. I don't see anything that's broken by this PR - nor should it.

**Comment by pllim** [comment]
Re: 404 -- That is weird because the link works for me.

**Comment by pllim** [comment]
Thanks for the quick review/merge!

**Pull request merged**